### PR TITLE
Johny fix bug on create folder for tasks with subtasks

### PR DESCRIPTION
**Pull request by johny4437**
Fixe  bug ( PRIORITY MEDIUM) - 1

# Description
Creating sub-tasks in a WBS doesn’t create a new folder and makes the tasks uneditable
Logged in as Admin → Other Links → Projects → AAA USE THIS PROJECT → Click WBS Icon → Choose “Jae's WBS - USE THIS” → Choose task not in folder → Edit → Add Task → Should create folder out of parent and including new task

## Mainly changes explained:
1 - Creating sub-tasks in a WBS  create a new folder.
2 -  Make subtasks editable and can be deleted.
3 - Now coming  from the dashboard is able to see the tasks that are in the same folder, edit those tasks and delete it.

## How to test:
1 - check into current branch
2 - do `npm install` and `...` to run this PR locally
3 - log as admin user
4 - Go to dashboard→ Other Links→ Projects→ chose the Project "AAA USE THIS PROJECT"→ click  "[ Jae's WBS - USE THIS]"
→ Add a task → Create a subtask for this task
Captura de Tela (71)
Captura de Tela (72)

 

## Note:
If you look at the bug list medium priority 1 you will be able to see a video where Jae is explaining about the  bug and 
 there is a video on my dropbox folder explaining about this PR please watch boths before you test it. See the links of the videos below:
Jae explaining about the bug:
 [Video - 1](https://www.dropbox.com/s/xm3rs215fl9a9uh/Sub-tasks%20Folder%20Bug.mov?dl=0)
My video explaining how to test the PR :
[Video - 2](https://www.dropbox.com/s/m8pweyyuvzwb4fq/how_to_test_the_pull_request.mp4?dl=0)

### CODE REVIEW

**Comment by EvianTan** [comment]
Hey @johny4437 , 

Awesome job on this PR and the above video!
✅ Verified changes:
1. "copy and add tasks" way created task will show in a related folder
2. tasks can be edited from the "same tasks" folder page
3. tasks can be deleted from the "same tasks" folder page

💡 Things can be improved
1. When I added and then delete the task "10.1" I just copied and created from task 10, on the WBS page it still shows task 10 as a folder. Is is the expected behavior? Once all subtasks got deleted, should be the mother task shown as non-folder one?
image

File: src/actions/task.js [hunk 1]
```diff
@@ -46,6 +54,7 @@ export const addNewTask = (newTask, wbsId) => async (dispatch, getState) => {
     _id = res.data._id;
     status = res.status;
     task = res.data;
+    console.log(task);
```
Comment (EvianTan) [inline src/actions/task.js:57]: Please delete this line. We don't allow any debug lines on our production code base.

File: src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx [hunk 2]
```diff
@@ -355,6 +369,7 @@ const AddTaskModal = (props) => {
     const res = props.allProjects.projects.filter(obj => obj._id === props.projectId)[0];
     if (res){
       setProjectCategory(res.category);
+      setClassification(res.category);
```
Comment (EvianTan) [inline src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:372]: Could you explain why here need this line?

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx [hunk 3]
```diff
@@ -26,6 +26,8 @@ const Task = props => {
   useEffect(() => {
     setIsCopied(false);
   }, [1]);
+
+  // console.log('TASKS ===>>>', props);
```
Comment (EvianTan) [inline src/components/Projects/WBS/WBSDetail/Task/Task.jsx:30]: Also as above, please remove this commented debug output

File: src/components/SummaryBar/SummaryBar.jsx [hunk 4]
```diff
@@ -83,6 +83,7 @@ const SummaryBar = props => {
   useEffect(() => {
     loadUserProfile();
     getUserTask();
+    // getWeeklySummary();
```
Comment (EvianTan) [inline src/components/SummaryBar/SummaryBar.jsx:86]: Please remove this line.

**Comment by johny4437** [comment]
I made the requested changes,  please verify.

File: src/actions/task.js [hunk 5]
```diff
@@ -3,15 +3,20 @@
  * Author: Henry Ng - 03/20/20
  ********************************************************************************/
 import axios from 'axios';
-import { fetchTeamMembersTaskSuccess, fetchTeamMembersTaskBegin, fetchTeamMembersTaskError } from 'components/TeamMemberTasks/actions';
```
Comment (EvianTan) [inline src/actions/task.js:6]: All formatting changes in this file, don't need to review.

**Comment by EvianTan** [comment]
@johny4437 
Thanks for the quick rework. There are still two things need to be addressed before my approval. Please check the "Things can be improved" part in my last comment.

**Comment by Filipe-Oliv** [comment]
Following up on this.

**Comment by johny4437** [comment]
Guys since I am getting a lot of problems with this PR I will close this one and I will create another for the frontend with more clean code, @EvianTan, @juneluo, @Filipe-Oliv

**Comment by kamtoeddy** [comment]
Hi @johny4437,

I got this build error:
Screenshot (480)

**Comment by one-community** [comment]
Is it ok to delete this branch? Please do, if it is.

**Pull request closed without merging**